### DISCUSSION TITLE
Dependabot-like workflow for container image automation

### DISCUSSION THREAD

**Discussion by sa-spag** [fluxcd/flux2#2138] [Proposals]
👋 Hello,

[As discussed on Slack](https://cloud-native.slack.com/archives/CLAJ40HV3/p1637770273038700), it could be handy to be able to achieve a Dependabot-like workflow for container image automation. The goal is to have pull requests acting as a manual gate for delivery.

[This document](https://fluxcd.io/docs/use-cases/gh-actions-auto-pr/) describes a setup which is close, but since [ImageUpdateAutomation's push spec](https://fluxcd.io/docs/components/image/imageupdateautomations/#push) only allows a statically-named branch to be specified, it seems it is not currently doable. In fact, as a single ImageUpdateAutomation applies to a whole namespace, pull requests created from Flux's owned branches could include unrelated changes, making them kind of unusable.

A set of solutions implies generating branch names (e.g. based on the updated manifest path, a random hash, etc.) instead of specifying a known branch name.

**Comment by iainmcgin** [comment]
A more recent related discussion: https://github.com/fluxcd/flux2/discussions/3135

**Comment by sa-spag** [comment]
Such workflow is achievable with Argo CD Image Updater, which allows target branch name templates ([see the documentation](https://argocd-image-updater.readthedocs.io/en/stable/basics/update-methods/#specifying-a-separate-base-and-commit-branch)), e.g. by annotating Argo CD Applications with:
```
argocd-image-updater.argoproj.io/git-branch: :image-updater-{{.SHA256}}
```
